core/src/test/scala/org/apache/spark/executor/CoarseGrainedExecutorBackendSuite.scala

```diff
@@ -302,7 +302,12 @@ class CoarseGrainedExecutorBackendSuite extends SparkFunSuite
           resourceProfile = ResourceProfile.getOrCreateDefaultProfile(conf))
       assert(backend.taskResources.isEmpty)
 
-      val taskId = 1000000
+      // Ensure thread-safety (https://issues.apache.org/jira/browse/SPARK-45227)
+      assert(backend.taskResources.isInstanceOf[
+        ConcurrentMap[Long, Map[String, ResourceInformation]]
+      ])
```

Comment:
Do you want to drop this `assert` @xiongbo-sjtu  ?

Reply:
The above `assert` logic was added so that the thread-safety data structure (i.e., CMH) will not be unintentionally refactored away later.  That's why I think that it's better for us to keep the logic here.

Reply:
We should add a comment on the field about thread safety, and remove this assertion.

Reply:
In the spirit of "disagree and commit", I've removed the assert statement and updated the pull request.  

Please merge this fix to master. If possible, please help patch v3.3.1 and above.  We have been waiting for the fix to get merged, before pulling the trigger to sync up our internal repo with the open-source repo.